### DISCUSSION TITLE
Stream by matches without or with an empty prefix

### DISCUSSION THREAD

**Discussion by ivanlegin** [ravendb/ravendb#16155] [Q&A]
```
string idPrefix = "Orders/";
string matches = "*25-A|77?-A";

// Filter streamed results by the passing 'prefix' and an optional 'matches' string
IEnumerator<StreamResult<Order>> streamResults = session.Advanced.Stream<Order>(idPrefix, matches);

while (streamResults.MoveNext())
{
    // Documents that will be returned are only those matching the following:
    // * Document ID starts with "Orders/"
    // * The rest of the ID (after prefix) must match the 'matches' string
    // e.g. "Orders/325-A" or Orders/772-A", etc.
    
    StreamResult<Order> currentResult = streamResults.Current;
    Order order = currentResult.Document;
}
```
Why can't we look for matches without prefixes here, but exactly IDs like:
```
string matches = "Orders-25-A|Orders-77-A";
IEnumerator<StreamResult<Order>> streamResults = session.Advanced.Stream<Order>(null | string.empty, matches);
```
If the prefix is ​​null or an empty string, the query flow ignores matches, why?
I want that method because no automatic indexing is created when streaming results by prefix.
That way I can stream unstale results, but a null or empty string for the prefix argument disables the filter by matches and returns any document

**Comment by ayende** [comment]
If you already know the ids of the documents in question, why are you not loading them directly? No need for streaming.
For that matter, the whole point of the `startsWith` is that you have.. a starts with, see:

https://github.com/ravendb/ravendb/blob/v5.4/src/Raven.Server/Documents/Handlers/DocumentHandler.cs#L175

If you don't provide a prefix, it means that we need to scan through all documents. In this case, better to use an index.

**Comment by ivanlegin** [comment]
If I have a large number of IDs, stream and processing one at a time seems to be a better option than loading them all into memory
But only way is stream from index(query), not directly by ids ?
`session.Stream(query<EntType>.Where(d=>d.Id.In(listIds))`

**Reply by ayende** [reply]
What is _large number of ids_ ?

**Reply by ivanlegin** [reply]
A string collection of thousands of id elements
